File: Conflict.py
import json
import csv

#creating a variable that contains conflicts from Asia and Africa only in the year 2000
World_Year = []
Year_2000 = []
Asia_Africa_2000 = [] # list of lists - each element is a list for a conflict, e.g. [region1, type1]
#making a list that includes the year on top of region and type of violence, including every line of the data
with open ('conflict_data_full_lined.json', encoding = 'utf-8-sig') as file:
    conflict = json.load(file)
    for line in conflict:
        World_Year.append([line['region'], line['type_of_violence'], line['year']])
#making a list that includes all the data that shows conflicts that occured in 2000, with information on their region and type of violence
with open ('conflict_data_full_lined.json', encoding = 'utf-8-sig') as file:
    conflict = json.load(file)
    for line in conflict:
        Year_2000.append([line['region'], line['type_of_violence']])
        if (line['region'] == 'Asia') and (line['year'] == 2000):
            Asia_Africa_2000.append([line['region'], line['type_of_violence']])
        if (line['region'] == 'Africa') and (line['year'] == 2000):
            Asia_Africa_2000.append([line['region'], line['type_of_violence']])
#print(Asia_Africa_2000)
#making a csv with region and type of violence
with open ('Asia_Africa_2000.csv','w') as file:
    filewriter = csv.writer(file)
    filewriter.writerow(['region', 'type_of_violence'])
    for line in Asia_Africa_2000:
        filewriter.writerow(line)
with open ('World_Year.csv','w') as file:
    filewriter = csv.writer(file)
    filewriter.writerow(['region', 'type_of_violence', 'year'])
    for line in World_Year:
        filewriter.writerow(line)
with open ('2000.csv','w') as file:
    filewriter = csv.writer(file)
    filewriter.writerow(['region', 'type_of_violence', 'year'])
    for line in Year_2000:
        filewriter.writerow(line)

File: conflict.R
library(tidyverse)

conflicts <- read.csv('Asia_Africa_2000.csv', stringsAsFactors = F)
conflicts2 <- read.csv('2000.csv', stringsAsFactors = T)
conflicts2 <- ordered(conflicts2,
                      levels = c("Africa", "Americas", "Asia", "Europe", "Middle East"))
                      #labels = c("AF", "AM", "A", "EU", "ME"))
conflicts3 <- read.csv('World_Year.csv', stringsAsFactors = F)
glimpse(conflicts)
conflicts

#This shows the count without separating into 3 types of conflict
ggplot()+
  geom_bar(data = conflicts, aes(x=region))

#graph that shows the different types of conflicts the 2 continents
ggplot()+
  geom_bar(data = conflicts, aes(x=region, color = 'red')) +
  facet_grid(~type_of_violence)+
  ggtitle('Type of Conflicts in Africa and Asia in 2000') +
  xlab('Continent') +
  ylab('Number of Incidents')

#graph that shows conflicts of all continents in all years, faceted by conflict type
ggplot()+
  geom_bar(data = conflicts2, aes(x=region, color = 'red')) +
  facet_grid(~type_of_violence)+
  ggtitle('Type of Conflicts in the world from 1989 to 2015') +
  xlab('Continent') +
  ylab('Number of Incidents') +
  theme(legend.position = "none") +
  scale_x_discrete(
    labels = c("AF", "AM", "A", "EU", "ME")
)

#creating a line graph that separates by region, showing the changes in the number of conflicts over time
conflict_lines <- conflicts3 %>%
    group_by(year, region) %>%
    summarise(n_conflicts = n())

ggplot()+
  geom_line(data = conflict_lines, aes(x=year, y = n_conflicts, group = region, color = region)) +
  ggtitle('Number of conflicts around the world by year by continent') +
  xlab('Year')+
  ylab('Number of Incidents')
